delira/utils/config.py

```diff
@@ -1,13 +1,509 @@
+        encoder_cls : :class:`Encoder`, optional
+            trasforms config to a format which can be formatted by the
+            `formatter`, by default Encoder
+        kwargs:
+            additional keyword arguments passed to `formatter`
+        """
+        self._timestamp = now()
+        encoded_self = encoder_cls().encode(self)
+        with open(path, "w") as f:
+            formatter(encoded_self, f, **kwargs)
+
+    def dumps(self, formatter=yaml.dump, encoder_cls=Encoder, **kwargs):
+        """
+        Create a loadable string representation from the config and
+        add time stamp to config
+
+        Parameters
+        ----------
+        formatter : callable, optional
+            defines the format how the config is saved, by default yaml.dump
+        encoder_cls : :class:`Encoder`, optional
+            trasforms config to a format which can be formatted by the
+            `formatter`, by default Encoder
+        kwargs:
+            additional keyword arguments passed to `formatter`
+        """
+        self._timestamp = now()
+        encoded_self = encoder_cls().encode(self)
+        return formatter(encoded_self, **kwargs)
+
+    @classmethod
+    def load(cls, path, formatter=yaml.load, decoder_cls=Decoder, **kwargs):
+        """
+        Load config from a file
+
+        Parameters
+        ----------
+        path : str
+            path to file
+        formatter : callable, optional
+            defines the format how the config is saved, by default yaml.dump
+        encoder_cls : :class:`Encoder`, optional
+            trasforms config to a format which can be formatted by the
+            `formatter`, by default Encoder
+        kwargs:
+            additional keyword arguments passed to `formatter`
+        """
+        with open(path, "r") as f:
+            decoded_format = formatter(f, **kwargs)
+        decoded_format = decoder_cls().decode(decoded_format)
+        if not isinstance(decoded_format, cls):
+            decoded_format = cls(decoded_format)
+        return decoded_format
+
+    @classmethod
+    def loads(cls, data, formatter=yaml.load, decoder_cls=Decoder, **kwargs):
+        """
+        Load config from a string
+
+        Parameters
+        ----------
+        formatter : callable, optional
+            defines the format how the config is saved, by default yaml.dump
+        encoder_cls : :class:`Encoder`, optional
+            trasforms config to a format which can be formatted by the
+            `formatter`, by default Encoder
+        kwargs:
+            additional keyword arguments passed to `formatter`
+        """
+        decoded_format = formatter(data, **kwargs)
+        decoded_format = decoder_cls().decode(decoded_format)
+        if not isinstance(decoded_format, cls):
+            decoded_format = cls(decoded_format)
+        return decoded_format
+
+    @classmethod
+    def create_from_dict(cls, value, deepcopy=False):
```

Comment:
What's the difference from calling `Config.create_from_dict(a_dict)` to simply `Config(a_dict)`? If there is a difference, this should be stated somethere, else this function is not necessary

Reply:
The is no difference. I preferred the syntax (it is also consistent with the `create_from_argparse` ) but ran into problems, because dicts need to support conversion of input parameters inside the `__init__` function. That's why there are two ways to do the same thing. 
If you prefer removing it, I can do that.

Reply:
I'm fine with keeping it, but shouldn't we simply call the init here after the assert with the value dict?

Reply:
You are right, I will change that.